### ISSUE TITLE
Expected dict, got <coroutine object ClusterNode.run

### ISSUE THREAD

**Issue by antoremin** [danielleyahalom/company-researcher#3]
When I run the app, I get an error at cluster node:

```
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/uvicorn/protocols/websockets/websockets_impl.py", line 255, in run_asgi
    result = await self.app(self.scope, self.asgi_receive, self.asgi_send)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/starlette/applications.py", line 123, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/starlette/middleware/errors.py", line 151, in __call__
    await self.app(scope, receive, send)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/starlette/routing.py", line 758, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/starlette/routing.py", line 778, in app
    await route.handle(scope, receive, send)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/starlette/routing.py", line 375, in handle
    await self.app(scope, receive, send)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/starlette/routing.py", line 98, in app
    await wrap_app_handling_exceptions(app, session)(scope, receive, send)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/starlette/routing.py", line 96, in app
    await func(session)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/fastapi/routing.py", line 360, in app
    await dependant.call(**values)
  File "/Users/antoneremin/workspace/experiments/company-researcher/app.py", line 37, in websocket_endpoint
    await graph.run(progress_callback=progress_callback)
  File "/Users/antoneremin/workspace/experiments/company-researcher/backend/graph.py", line 92, in run
    async for s in graph.astream(self.state, thread, stream_mode="values"):
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/langgraph/pregel/__init__.py", line 1524, in astream
    async for _ in runner.atick(
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/langgraph/pregel/runner.py", line 130, in atick
    await arun_with_retry(t, retry_policy, stream=self.use_astream)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/langgraph/pregel/retry.py", line 102, in arun_with_retry
    await task.proc.ainvoke(task.input, config)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/langgraph/utils/runnable.py", line 455, in ainvoke
    input = await asyncio.create_task(coro)
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/langgraph/utils/runnable.py", line 226, in ainvoke
    ret = await coro
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/langgraph/pregel/write.py", line 99, in _awrite
    self.do_write(
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/langgraph/pregel/write.py", line 125, in do_write
    values = [
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/langgraph/pregel/write.py", line 126, in <listcomp>
    write.mapper(write.value) if write.mapper is not None else write.value
  File "/Users/antoneremin/.pyenv/versions/3.9.11/lib/python3.9/site-packages/langgraph/graph/state.py", line 545, in _get_state_key
    raise InvalidUpdateError(msg)
langgraph.errors.InvalidUpdateError: Expected dict, got <coroutine object ClusterNode.run at 0x16c31cd40>
For troubleshooting, visit: https://python.langchain.com/docs/troubleshooting/errors/INVALID_GRAPH_NODE_RETURN_VALUE
```

**Comment by lattaai13** [comment]
Hello, I tried to solve the issue.

This is what I did:

Modified the ClusterNode.run method in backend/nodes/cluster.py to return a dictionary directly instead of being an async function. This change resolves the InvalidUpdateError by ensuring the method returns the expected dict type.

You can review changes in this commit: https://github.com/lattaai13/danielleyahalom-company-researcher-3/commit/e1dda416ed1ee5606803a0615dada561f1be0e93.


> [!CAUTION]
> Disclaimer: The commit was created by Latta AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---
This issue was tried to solve for free by Latta AI - https://latta.ai/ourmission

If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by danielleyahalom** [comment]
The issue has been resolved in commit 8846da7. 

The fix addresses coroutine handling in Python 3.9 by using curried wrappers to preserve coroutine behavior and ensure nodes execute asynchronously with parameters like `websocket`.
